pytorch_lightning/trainer/trainer.py

```diff
@@ -1200,9 +1191,11 @@ def call_hook(self, hook_name: str, *args, **kwargs) -> Any:
         with self.profiler.profile(hook_name):
 
             # first call trainer hook
-            if hasattr(self, hook_name):
+            if hook_name not in ("setup", ) and hasattr(self, hook_name):
```

Comment:
Ideally, connectors should also be called using `call_hook`.

Reply:
Opened #8506 to discuss